BaseParser.inc.php

```diff
@@ -118,8 +118,15 @@ private function _ensureMetadataIsValidAndParse(): self
 
 		// Checks whether the loaded document is supported by the parser (the doctype should match)
 		$docType = $this->_document->doctype;
-		$supportedDocType = $this->getDocType();
-		if ([$docType->systemId, $docType->publicId, $docType->name] != [$supportedDocType->systemId, $supportedDocType->publicId, $supportedDocType->name]) {
+		$supportedDocTypes = $this->getDocType();
+		$found = false;
+		foreach ($supportedDocTypes as $supportedDocType) {
+			if ([$docType->systemId, $docType->publicId, $docType->name] == [$supportedDocType->systemId, $supportedDocType->publicId, $supportedDocType->name]) {
```

Comment:
I didn't like much to use such strict versions, so I was thinking about changing this part to be more relaxed :)
In fact, I wanted to refactor this plugin to work in a totally different way... Errrr, to answer for "data requests" instead of calling PKP code by itself, then I think it would be generic enough while also killing duplicated code 🤔

What about this NLM format? Is it generic enough to be merged in with the JATS?

Reply:
[NLM DTD 3.0 became JATS 1.0](https://dtd.nlm.nih.gov/), so adding support for the additional doctype effectively adds support for NLM DTD 3.0.  There have been some minor variations over time: for example, the authorship relationship can be captured withing the `contrib-group` or the `contrib` entities, in NLM 3.x or JATS 1.x.  The node containing the author's names was renamed from `string-name` to `name` at some point.

Reply:
By the way, +1 to relaxing the DTD check here.

Reply:
Thanks @ctgraham! So I'll merge it as it is, then when I get some free time I might revisit this plugin :)

Reply:
If you do want to consider a more comprehensive refactor of the plugin for broader DTD support, one approach could be to have this plugin extend the `NativeImportExportPlugin`.  An XSLT transform could convert the JATS / NLM / aPlusPlus XML to the native XML, and just hand it off to the NativeImportExportPlugin's processing.  The structure for batch ingest which you've setup and the mapping of the known DTDs would be the work of this plugin; the details of actually interfacing to the OJS API could be left to the existing native plugin.

Reply:
I fully agree, having a single point to import things is the best option to avoid having a broken toy.

I've created an issue referencing these comments, so I can remember to take a look later and redirect any kind of improvement effort towards the native plugin.